ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -120,70 +124,23 @@ public Schema getResultSchema() {
     return sourceNames;
   }
 
-  public void close() {
-    kafkaStreams.close();
-    if (kafkaStreams.state() == KafkaStreams.State.NOT_RUNNING) {
-      kafkaStreams.cleanUp();
-      kafkaTopicClient.deleteInternalTopics(queryApplicationId);
-    } else {
-      log.error("Could not clean up the query with application id: {}. Query status is: {}",
-                queryApplicationId, kafkaStreams.state());
-    }
-    queryStateListener.ifPresent(QueryStateListener::close);
-    StreamsErrorCollector.notifyApplicationClose(queryApplicationId);
-  }
-
-  private Set<String> getInternalSubjectNameSet(final SchemaRegistryClient schemaRegistryClient) {
-    try {
-      final String suffix1 = KsqlConstants.STREAMS_CHANGELOG_TOPIC_SUFFIX
-                             + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
-      final String suffix2 = KsqlConstants.STREAMS_REPARTITION_TOPIC_SUFFIX
-                             + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX;
-
-      return schemaRegistryClient.getAllSubjects().stream()
-          .filter(subjectName -> subjectName.startsWith(getQueryApplicationId()))
-          .filter(subjectName -> subjectName.endsWith(suffix1) || subjectName.endsWith(suffix2))
-          .collect(Collectors.toSet());
-    } catch (final Exception e) {
-      // Do nothing! Schema registry clean up is best effort!
-      log.warn("Could not clean up the schema registry for query: " + queryApplicationId, e);
-    }
-    return new HashSet<>();
+  public boolean hasEveryBeenStarted() {
+    return everyStarted;
   }
 
+  public void close() {
+    kafkaStreams.close();
 
-  public void cleanUpInternalTopicAvroSchemas(final SchemaRegistryClient schemaRegistryClient) {
-    getInternalSubjectNameSet(schemaRegistryClient).forEach(subjectName -> {
-      try {
-        schemaRegistryClient.deleteSubject(subjectName);
-      } catch (final Exception e) {
-        log.warn("Could not clean up the schema registry for query: " + queryApplicationId
-                 + ", topic: " + subjectName, e);
-      }
-    });
-  }
-
-  @Override
-  public boolean equals(final Object o) {
-    if (!(o instanceof QueryMetadata)) {
-      return false;
-    }
-
-    final QueryMetadata that = (QueryMetadata) o;
+    kafkaStreams.cleanUp();
```

Comment:
Previously this `cleanUp()` call happened after the check for whether the streams state is `NOT_RUNNING`, but in the new version the call is always made, even if the streams state isn't `NOT_RUNNING`. From looking at the docs this seems fine, but wanted to draw attention to it since it's just about the only functional change I see.

Reply:
Yeah, I did notice that and originally had it wrapped.  Then I thought I'd just check the docs and code and it looks good to be called outside of the check. 

And yes, I think this is the only functional change.